tests/kokkos-based/helpers.hpp

```diff
@@ -166,60 +195,216 @@ bool is_same_vector(
   return is_same_vector(make_mdspan(v1), make_mdspan(v2));
 }
 
-template <typename RealValue>
-KOKKOS_INLINE_FUNCTION
-RealValue scalar_diff(RealValue v1, RealValue v2) {
-  return std::abs(v2 - v1);
+template <typename ElementType1,
+          typename LayoutPolicy1,
+          typename AccessorPolicy1,
+          typename ElementType2,
+          typename LayoutPolicy2,
+          typename AccessorPolicy2>
+auto vector_abs_diff(
+    mdspan<ElementType1, extents<dynamic_extent>, LayoutPolicy1, AccessorPolicy1> v1,
+    mdspan<ElementType2, extents<dynamic_extent>, LayoutPolicy2, AccessorPolicy2> v2)
+{
+  using RetType = decltype(scalar_abs_diff(v1[0], v2[0])); // will be same for views
+  const auto size = v1.extent(0);
+  if (size != v2.extent(0))
+    return std::numeric_limits<RetType>::max(); // very, very different
+  const auto v1_view = KokkosKernelsSTD::Impl::mdspan_to_view(v1);
+  const auto v2_view = KokkosKernelsSTD::Impl::mdspan_to_view(v2);
+  RetType diff = static_cast<RetType>(0);
+  Kokkos::parallel_reduce(size,
+    KOKKOS_LAMBDA(const std::size_t i, RetType &diff){
+        const RetType d = scalar_abs_diff(v1_view[i], v2_view[i]);
+        if (d > diff) {
+          diff = d;
+        }
+	    }, Kokkos::Max<RetType>(diff));
+  return diff;
 }
 
-template <typename T>
-T scalar_diff(const std::complex<T> &v1, const std::complex<T> &v2) {
-  const auto dr = scalar_diff(v1.real(), v2.real());
-  const auto di = scalar_diff(v1.imag(), v2.imag());
-  return std::max(dr, di);
+template <typename ElementType1,
+          typename LayoutPolicy,
+          typename AccessorPolicy,
+          typename ElementType2>
+auto vector_abs_diff(
+    mdspan<ElementType1, extents<dynamic_extent>, LayoutPolicy, AccessorPolicy> v1,
```

Comment:
Using `dynamic_extent` is fine here, but would you consider instead templating on the extent, like the following?
```c++
template <typename ElementType1,
          std::size_t Extent,
          typename LayoutPolicy,
          typename AccessorPolicy,
          typename ElementType2>
auto vector_abs_diff(
    mdspan<ElementType1, extents<Extent>, LayoutPolicy, AccessorPolicy> v1,
    const std::vector<ElementType2>& v2) { /* ... */ }
```
That would make it possible to compare a `vector` with an `mdspan` of compile-time extent.

Reply:
Thank you for pointing this out! Fixed on https://github.com/kokkos/stdBLAS/pull/193/commits/860d497d1f77de874de43e69af7bfea608e23e2b.